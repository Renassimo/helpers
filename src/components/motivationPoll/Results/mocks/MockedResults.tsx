const MockedResults = jest.fn(
  ({ onDownloadPdf }: { onDownloadPdf: () => void }) => (
    <div>Mocked Results, onDownloadPdf: {!!onDownloadPdf}</div>
  )
);

export default MockedResults;
