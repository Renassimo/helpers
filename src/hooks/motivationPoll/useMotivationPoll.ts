import { useContext } from 'react';

import MotivationPollContext from 'src/contexts/motivationPoll';

const useMotivationPoll = () => useContext(MotivationPollContext);

export default useMotivationPoll;
