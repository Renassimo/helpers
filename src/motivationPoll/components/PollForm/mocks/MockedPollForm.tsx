const MockedPollForm = jest.fn(() => <div>Mocked Poll Form</div>);

export default MockedPollForm;
