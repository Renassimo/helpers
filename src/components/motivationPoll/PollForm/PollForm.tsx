import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import Description from '@/components/motivationPoll/Description';
import StarterForm from '@/components/motivationPoll/StarterForm';
import QuestionForm from '@/components/motivationPoll/QuestionForm';
import FinishForm from '@/components/motivationPoll/FinishForm';

import Box from '@mui/material/Box';

const PollForm = () => {
  const { isStarted, isFinished, currentQuestion } = useMotivationPoll();

  return (
    <Box>
      {!isFinished && <Description />}
      {!isStarted && <StarterForm />}
      {currentQuestion && <QuestionForm />}
      {isFinished && <FinishForm />}
    </Box>
  );
};

export default PollForm;
