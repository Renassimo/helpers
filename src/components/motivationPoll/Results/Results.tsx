import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import ResultsTable from '@/components/motivationPoll/ResultsTable';
import ResultsChart from '@/components/motivationPoll/ResultsChart';

import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';

const Results = ({ onDownloadPdf }: { onDownloadPdf: () => void }) => {
  const { name } = useMotivationPoll();
  return (
    <Box>
      <Typography component="h1" variant="h5" textAlign="center" my={2}>
        Motivation Test Results for {name}:{' '}
        <Button onClick={onDownloadPdf}>Save as PDF</Button>
      </Typography>
      <ResultsTable />
      <ResultsChart />
    </Box>
  );
};

export default Results;
