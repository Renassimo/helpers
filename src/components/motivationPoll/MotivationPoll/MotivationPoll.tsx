import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import PollForm from '@/components/motivationPoll/PollForm';
import Results from '@/components/motivationPoll/Results';

import Box from '@mui/material/Box';

const MotivationPoll = ({ onDownloadPdf }: { onDownloadPdf: () => void }) => {
  const { results } = useMotivationPoll();
  return (
    <Box>
      {results ? <Results onDownloadPdf={onDownloadPdf} /> : <PollForm />}
    </Box>
  );
};

export default MotivationPoll;
