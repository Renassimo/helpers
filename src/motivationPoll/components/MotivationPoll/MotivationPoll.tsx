import useMotivationPoll from '@/motivationPoll/hooks/useMotivationPoll';

import PollForm from '@/motivationPoll/components/PollForm';
import Results from '@/motivationPoll/components/Results';

import Box from '@mui/material/Box';

const MotivationPoll = ({ onDownloadPdf }: { onDownloadPdf: () => void }) => {
  const { results } = useMotivationPoll();
  return (
    <Box>
      {results ? <Results onDownloadPdf={onDownloadPdf} /> : <PollForm />}
    </Box>
  );
};

export default MotivationPoll;
