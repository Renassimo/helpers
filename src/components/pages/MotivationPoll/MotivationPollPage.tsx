import MotivationPollProvider from '@/providers/motivationPoll/MotivationPollProvider';

import usePdfDownload from '@/hooks/common/usePdfDownload';

import { MotivationPollApiData } from '@/types/motivationPoll';

import PageTemplate from '@/components/templates/PageTemplate';
import MotivationPoll from '@/components/motivationPoll/MotivationPoll';
import LocaleLinks from '@/common/components/LocaleLinks';

const MotivationPollPage = ({
  data,
  locales,
  locale,
}: {
  data: MotivationPollApiData;
  locales: string[];
  locale: string;
}) => {
  const { pdfRef, onDownloadPdf } = usePdfDownload('motivation-poll-results');
  return (
    <MotivationPollProvider data={data}>
      <div ref={pdfRef}>
        <PageTemplate title="Motivation Test" pages={[]}>
          <MotivationPoll onDownloadPdf={onDownloadPdf} />
          <LocaleLinks
            locale={locale}
            locales={locales}
            href="/motivationTest"
          />
        </PageTemplate>
      </div>
    </MotivationPollProvider>
  );
};

export default MotivationPollPage;
