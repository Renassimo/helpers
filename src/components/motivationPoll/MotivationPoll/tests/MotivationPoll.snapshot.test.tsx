import renderWithTheme from '@/tests/helpers';

import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import PollForm from '@/components/motivationPoll/PollForm';
import Results from '@/components/motivationPoll/Results';

import MockedPollForm from '@/components/motivationPoll/PollForm/mocks';
import MockedResults from '@/components/motivationPoll/Results/mocks';

import MotivationPoll from '../MotivationPoll';

jest.mock('@/hooks/motivationPoll/useMotivationPoll');
jest.mock('@/components/motivationPoll/PollForm');
jest.mock('@/components/motivationPoll/Results');

describe('MotivationPoll snapshot', () => {
  const mockedOnDownloadPdf = jest.fn();

  beforeEach(() => {
    (useMotivationPoll as jest.Mock).mockImplementation(() => ({
      results: null,
    }));
    (PollForm as jest.Mock).mockImplementation(MockedPollForm);
    (Results as jest.Mock).mockImplementation(MockedResults);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders PollForm', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <MotivationPoll onDownloadPdf={mockedOnDownloadPdf} />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });
  describe('when has results', () => {
    beforeEach(() => {
      (useMotivationPoll as jest.Mock).mockImplementation(() => ({
        results: [],
      }));
    });
    test('renders Results', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(
        <MotivationPoll onDownloadPdf={mockedOnDownloadPdf} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });
});
