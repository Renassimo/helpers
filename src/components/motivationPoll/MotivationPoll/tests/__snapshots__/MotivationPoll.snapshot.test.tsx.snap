// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotivationPoll snapshot renders PollForm 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Poll Form
      </div>
    </div>
  </div>
</body>
`;

exports[`MotivationPoll snapshot when has results renders Results 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Results, onDownloadPdf: 
      </div>
    </div>
  </div>
</body>
`;
